### PR TITLE
🎉 Init project

### PR DESCRIPTION
**Pull request by Ic3Sandy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+.env
+dist/
```
Comment (thebevrishot) [inline .gitignore:3]: please add .vscode/

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (thebevrishot) [inline .vscode/launch.json:1]: we dont have to have this on repo

File: app/index.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { Context, PrivkeyAccount, Client, RPCClient } from "firovm-sdk";
+import tokens from "./tokens.json";
+import "dotenv/config";
+
+const app: Express = express();
+const port = Number(process.env.PORT) || 8123;
+const bind = process.env.BIND || "0.0.0.0";
+const network = process.env.NETWORK || "Regtest"; // ["Mainnet", "Testnet", "Regtest"]
+const RPCURL = process.env.RPCURL || "http://guest:guest@127.0.0.1:8545";
+const PRIVKEY = process.env.PRIVKEY || "";
```
Comment (thebevrishot) [inline app/index.ts:12]: Please move environments section to between line 5 and 7

File: app/index.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
+import express, { Express, Request, Response } from "express";
+import { fromHexAddress, getNetwork } from "./utils";
+import { Context, PrivkeyAccount, Client, RPCClient } from "firovm-sdk";
+import tokens from "./tokens.json";
+import "dotenv/config";
+
+const app: Express = express();
+const port = Number(process.env.PORT) || 8123;
+const bind = process.env.BIND || "0.0.0.0";
+const network = process.env.NETWORK || "Regtest"; // ["Mainnet", "Testnet", "Regtest"]
```
Comment (thebevrishot) [inline app/index.ts:10]: Configuration from environment should be all upper cases

File: app/index.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        account,
+        [
+          {
+            to: nativeAddress as string,
+            value: 100 * 1e8,
+          },
+        ],
+        { feePerKb: 1000 }
+      );
+      return res.status(201).send({ tx: txId });
```
Comment (thebevrishot) [inline app/index.ts:81]: Please move this to else section of `if (token && token.address !== "") {`

File: app/index.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    if (token && token.address !== "") {
+      const txId = await client.tokenTransfer(
+        account,
+        token.address,
+        nativeAddress,
+        BigInt(1e18)
```
Comment (thebevrishot) [inline app/index.ts:89]: transfer amount should coming with configuration for each token

File: app/index.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    const token = tokens[asset as TypeTokens];
+    if (token && token.address === "") {
+      const txId = await client.sendFrom(
+        account,
+        [
+          {
+            to: nativeAddress as string,
+            value: 100 * 1e8,
```
Comment (thebevrishot) [inline app/index.ts:76]: This should be configurable

**Review by thebevrishot — CHANGES_REQUESTED** [review]

File: app/tokens.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "FVM": {
+    "address": ""
+  }
```
Comment (thebevrishot) [inline app/tokens.json:4]: dont think we need FVM here

**Review by thebevrishot — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Ic3Sandy) [reply .gitignore:3]: ok kub

File: app/index.ts (hunk 3 above)
Comment (Ic3Sandy) [reply app/index.ts:13]: ok kub

[the remark this answers is not in the document]
Comment (Ic3Sandy) [reply app/index.ts:13]: ok kub

File: app/index.ts (hunk 4 above)
Comment (Ic3Sandy) [reply app/index.ts:11]: ok kub

File: app/index.ts [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  new Context().withNetwork(getNetwork(network.toLowerCase())),
+  PRIVKEY
+);
+const rpc = new RPCClient(RPCURL);
+const client = new Client(RPCURL);
+type TypeTokens = keyof typeof tokens;
+
+app.use(express.json());
+
+app.get("/", async (req: Request, res: Response) => {
```
[the remark this answers is not in the document]
Comment (Ic3Sandy) [reply app/index.ts]: i will remove it kub

File: app/index.ts (hunk 7 above)
Comment (Ic3Sandy) [reply app/index.ts:58]: ok kub

File: app/tokens.json (hunk 8 above)
Comment (Ic3Sandy) [reply app/tokens.json:4]: I used to tell what name of the native coin Kub

File: app/index.ts (hunk 6 above)
Comment (Ic3Sandy) [reply app/index.ts]: I will add config for this kub

File: docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    restart: always
+    command: bash -c "npm i && npx tsc && npm run start"
+    stdin_open: true
+    tty: true
+    working_dir: /app
+    volumes:
+      - ./app:/app
+    ports:
+      - 8123:8123
+      - 9229:9229
```
[the remark this answers is not in the document]
Comment (Ic3Sandy) [reply docker-compose.yml:16]: ok kub

File: app/index.ts [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    return res.status(404).send({ message: "Asset not found!" });
+  } catch (err) {
+    console.error(err);
+    return res.status(500).send({ message: (<any>err).message });
+  }
+});
+
+app.get("/assets", async (req: Request, res: Response) => {
```
[the remark this answers is not in the document]
Comment (Ic3Sandy) [reply app/index.ts:87]: I will create cache for this kub

File: app/models/reachLimit.ts [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+  expireTime: number;
+}
+
+export const reachLimitCollection =
+  database.collection<ReachLimit>("reachlimits");
+
+export const setReachLimit = async (ip: string, address: string) => {
+  await reachLimitCollection.updateOne(
+    { ip, address },
+    { $set: { expireTime: Date.now() + CacheTimes.Day } },
```
Comment (thebevrishot) [inline app/models/reachLimit.ts:27]: Could we config interval?

File: app/tokens.json (hunk 8 above)
Comment (thebevrishot) [reply app/tokens.json:5]: okay

**Review by thebevrishot — CHANGES_REQUESTED** [review]

File: app/models/reachLimit.ts (hunk 12 above)
Comment (Ic3Sandy) [reply app/models/reachLimit.ts:27]: yes kub, I will add config to env file kub

File: app/tokens.json (hunk 8 above)
Comment (Ic3Sandy) [reply app/tokens.json:7]: I moved native token config to env file kub

**Review by thebevrishot — APPROVED** [review]

**Pull request merged**